pysat/tests/classes/cls_instrument_property.py

```diff
@@ -0,0 +1,648 @@
+                                                  ('load', 'test_load_kwarg',
+                                                   'bright_light'),
+                                                  ('list_files',
+                                                   'test_list_files_kwarg',
+                                                   'sleep_tight'),
+                                                  ('list_remote_files',
+                                                   'test_list_remote_kwarg',
+                                                   'one_eye_open'),
+                                                  ('download',
+                                                   'test_download_kwarg',
+                                                   'exit_night')
+                                                  ])
+    def test_instrument_function_keywords(self, caplog, func, kwarg, val):
+        """Test if Instrument function keywords are registered by pysat.
+
+        Parameters
+        ----------
+        func : str
+            Function name to test.
+        kwarg : str
+            Keyword argument of function to modify.
+        val : str
+            Replacement value for modified kwarg.
+
+        """
+
+        with caplog.at_level(logging.INFO, logger='pysat'):
+            # Trigger load functions
+            self.testInst.load(date=self.ref_time)
+
+            # Refresh files to trigger other functions
+            self.testInst.files.refresh()
+
+            # Get remote file list
+            self.testInst.download_updated_files()
+
+        # Confirm kwargs made it where they should be
+        assert kwarg in self.testInst.kwargs[func]
+        assert self.testInst.kwargs[func][kwarg] == val
+
+        # Check if function under test can assign attributes, not all can
+        live_check = hasattr(self.testInst, kwarg)
+
+        if live_check:
+            # Confirm attribute value
+            assert getattr(self.testInst, kwarg) == val
+        else:
+            # Confirm value echoed to log for functions that can't assign
+            # attributes. Get log text.
+            captured = caplog.text
+
+            # Test for expected string
+            test_str = ''.join((kwarg, ' = ', str(val)))
+            assert captured.find(test_str) >= 0
+
+        return
+
+    @pytest.mark.parametrize("func, kwarg", [('clean', 'test_clean_kwarg'),
+                                             ('preprocess',
+                                              'test_preprocess_kwarg'),
+                                             ('load',
+                                              'test_load_kwarg'),
+                                             ('list_files',
+                                              'test_list_files_kwarg'),
+                                             ('list_files',
+                                              'test_list_files_kwarg'),
```

Comment:
Duplicated keyword test.

Reply:
`init` is missing from the list.

Reply:
Just tried to run this.  init doesn't work

Reply:
removed duplicate test.  Not including init since this doesn't work now.  Will revisit in #914.